[BUG] Certification test 5.7.1 fails

tomereli opened the pull request:
Controller certification tests 5.7.1 fails on fefec5f.

Logs can be found here: https://gitlab.com/prpl-foundation/prplMesh/-/jobs/401365002/artifacts/browse/logs/FAIL/

arnout commented:
![](https://github.trello.services/images/mini-trello-icon.png) [Check controller certification test failures](https://trello.com/c/iJIjpQJp/68-check-controller-certification-test-failures)

arnout commented:
The combined infra message is empty. I'll create a test_flows.sh script for this. It was missing, so it's not surprising that it breaks :-)

Commit:
controller: call finalize for certification cmdu

When we prepare the certification cmdu, but don't finalize it, it is
never used by the ucc listener. So add finalize calls in all places
where we create a certification cmdu.

Signed-off-by: Tomer Eliyahu <tomer.b.eliyahu@intel.com>

Commit:
tlvf: add CmduMessage::setMessageId()

Preparative commit.
Add a new API to set message id in the CMDU header which will be used to
force mid value by UCC listener when sending a preset certification
cmdu.

Also, while at it, fix all C style casts in this file.

Signed-off-by: Tomer Eliyahu <tomer.b.eliyahu@intel.com>

Commit:
ucc listener: fix wrong endianess when setting mid

If the certificaiton cmdu is preset and ready to be sent, the UCC
listener need to update the MID value in the CMDU header.
Since the header (and the entire CMDU) is already finalized, we can't
set the MID directly, since the endianess will be wrong.

Fix this by using the new CmduMessage::setMessageId() API which takes
care of endianess.

Fixes #669

Signed-off-by: Tomer Eliyahu <tomer.b.eliyahu@intel.com>

Commit:
ucc listener: add debug print when sending preset cmdu

Signed-off-by: Tomer Eliyahu <tomer.b.eliyahu@intel.com>

Commit:
agent: fix stubs for ap and link metric queries

Current stubs are redundant and missing log print so can't be grepped in
test_flow.sh. So remove redundant implementation and make sure each stub
has a log print with the received message type string.

Signed-off-by: Tomer Eliyahu <tomer.b.eliyahu@intel.com>

Commit:
test_flows.sh: add initial test_combined_infra_metrics

Currently the test fails since the controller does not receive the
responses from the agent, since they are stubs in the agent.
This will be fixed in
https://github.com/prplfoundation/prplMesh/pull/784, so for now the test
result is skipped so it will pass CI.

Signed-off-by: Tomer Eliyahu <tomer.b.eliyahu@intel.com>

rmelotte commented:
Rebased on master after https://github.com/prplfoundation/prplMesh/pull/766 was merged.

arnout commented on framework/tlvf/src/src/CmduMessage.cpp:
```diff
@@ -65,3 +65,12 @@ uint16_t CmduMessage::getMessageId()
+    auto cmduhdr = getCmduHeader();
+
+    if (cmduhdr->is_finalized())
+        swap_16((uint16_t &)mid);
```
This cast looks fishy. Why is it needed?

tomereli replied:
Copy-paste https://tse1.mm.bing.net/th?id=OGC.335506d103b360a3e59d56ddfa94fdb8&pid=Api&rurl=https%3a%2f%2fmedia.giphy.com%2fmedia%2f26xBN67Q1HzBDyU0w%2fsource.gif&ehk=1ypHjLI5Si28m2%2fbCzFpq6QelYRKn21j2RQUPFdV4wo%3d

vitalybu commented on agent/src/beerocks/slave/son_slave_thread.cpp:
```diff
@@ -4384,38 +4384,16 @@ bool slave_thread::handle_client_steering_request(Socket *sd, ieee1905_1::CmduMe
-        }
-        tlv_type = cmdu_tx.getNextTlvType();
-    }
+    LOG(DEBUG) << "Received AP_METRICS_QUERY_MESSAGE , mid=" << std::hex << int(mid);
```
```suggestion
    LOG(DEBUG) << "Received AP_METRICS_QUERY_MESSAGE, mid=" << std::hex << int(mid);
```

vitalybu commented on agent/src/beerocks/slave/son_slave_thread.cpp:
```diff
@@ -4384,38 +4384,16 @@ bool slave_thread::handle_client_steering_request(Socket *sd, ieee1905_1::CmduMe
-        }
-        tlv_type = cmdu_tx.getNextTlvType();
-    }
+    LOG(DEBUG) << "Received LINK_METRIC_QUERY_MESSAGE , mid=" << std::hex << int(mid);
```
```suggestion
    LOG(DEBUG) << "Received LINK_METRIC_QUERY_MESSAGE, mid=" << std::hex << int(mid);
```

rmelotte commented on framework/tlvf/src/src/CmduMessage.cpp:
```diff
@@ -65,3 +65,12 @@ uint16_t CmduMessage::getMessageId()
+    auto cmduhdr = getCmduHeader();
+
+    if (cmduhdr->is_finalized())
+        swap_16((uint16_t &)mid);
```
Is that a c-style cast (I know there are others in the file...)
~~Also, if you make setMessageId accept a reference instead, you don't need any cast.~~
Edit: the cast is not needed even if setMessageId doesn't take mid by reference. Does the caller of setMessageId need `mid` to be swapped?

tomereli replied:
I fixed it and got rid of the other C style casts in this file.

rmelotte reviewed (commented):
Please remove the c-style cast by any mean.
Otherwise, LGTM!

rmelotte reviewed (commented):
(Github insists on requiring a comment to mark approval)

rmelotte reviewed (commented):
[]
